gdk/common/exceptions/error_messages.py

```diff
@@ -7,12 +7,20 @@
     "No valid component recipe is found. Please include a valid recipe file of the component to build with default."
 )
 PROJECT_CONFIG_FILE_INVALID = "Project configuration file '{}' is invalid. Please correct its format and try again. Error: {} "
-PROJECT_RECIPE_FILE_INVALID = "Project recipe file '{}' is invalid. Please correct its format and try again. Error: {} "
+PROJECT_RECIPE_FILE_INVALID = (
+    "Project recipe file '{}' is invalid. Please correct its format and try again.\nFor more information regarding " +
+    "component recipes refer to the docs here: https://docs.aws.amazon.com/greengrass/v2/developerguide/" +
+    "component-recipe-reference.html\nError: {} "
+)
 SCHEMA_FILE_INVALID = (
     "The Greengrass recipe schema file has an unexpected error.\nPlease report it at " +
     "https://github.com/aws-greengrass/aws-greengrass-gdk-cli/issues if the issue persists.\nError details: {} "
 )
-BUILD_RECIPE_FILE_INVALID = "Built recipe file '{}' is invalid. Please correct its format and try again. Error: {} "
+BUILD_RECIPE_FILE_INVALID = (
+    "Built recipe file '{}' is invalid. Please correct its format and try again.\nFor more information regarding " +
+    "component recipes refer to the docs here: https://docs.aws.amazon.com/greengrass/v2/developerguide/" +
+    "component-recipe-reference.html\nError: {} "
+)
```

Comment:
Do we really need two different error messages? We can use the same message for both.

Reply:
I just wanted to make it clear that it isn't the original recipe file in their project for the publish case.

Reply:
What's the difference between a "project" and "built" recipe file? How can one be valid and the other invalid?

Reply:
The case I had in mind was if someone goes in the greengrass-build folder and edits the recipe to something that's invalid, then the publish command would be using an invalid recipe and so we go and validate it again. Shouldn't be a common case at all but if anything happens between the build and publish command the second validation during publish should still catch these errors.

Reply:
That seems unlikely. Providing the path to the file which was validated would be sufficient to clarify what's invalid.

Reply:
Not a blocker, but a quick change so: https://github.com/aws-greengrass/aws-greengrass-gdk-cli/pull/263